src/plugins/intel_gpu/src/graph/impls/ocl/concatenation.cpp

```diff
@@ -52,20 +52,31 @@ struct concatenation_impl : typed_primitive_impl_ocl<concatenation> {
         return make_unique<concatenation_impl>(*this);
     }
 
+    explicit concatenation_impl(const concatenation_impl& other) : parent(other),
+        _can_be_optimized(other._can_be_optimized) {}
+
     concatenation_impl(const concatenation_node& arg, const kernel_selector::kernel_data& kd) : parent(arg, kd) {
-        if (!_outer.can_be_optimized()) {
-            CLDNN_ERROR_NOT_EQUAL(_outer.id(),
+        if (!arg.can_be_optimized()) {
+            CLDNN_ERROR_NOT_EQUAL(arg.id(),
                                   "Input count",
-                                  _outer.inputs_count(),
+                                  arg.inputs_count(),
                                   "kds size",
                                   kd.kernels.size(),
                                   "Error - not enough kernels for concatenation");
         }
+
+        set_node_params(arg);
+    }
+
+    void set_node_params(const program_node& arg) override {
+        IE_ASSERT(arg.is_type<concatenation>());
+        const auto& node = arg.as<concatenation>();
+        _can_be_optimized = node.can_be_optimized();
```

Comment:
Note: I think we'll need to git rid of all the program_node dependency from primitive_impl create function in the future. 
At that time, all attributes should be from the kernel_param_impl->desc, and some additional program_node parameters such as can_be_optimized. etc, will be remaining. But for now, we can go with this.

Reply:
Okay, I'll improve it later based on your comments. Thank you!